src/backend/libc/mm/syscalls.rs

```diff
@@ -220,3 +229,36 @@ pub(crate) unsafe fn userfaultfd(flags: UserfaultfdFlags) -> io::Result<OwnedFd>
     }
     ret_owned_fd(userfaultfd(bitflags_bits!(flags)))
 }
+
+/// Locks all pages mapped into the address space of the calling process.
+///
+/// This includes the pages of the code, data and stack segment, as well as shared libraries,
+/// user space kernel data, shared memory, and memory-mapped files. All mapped pages are
+/// guaranteed to be resident in RAM when the call returns successfully;
+/// the pages are guaranteed to stay in RAM until later unlocked.
+#[inline]
+#[cfg(any(
+    linux_kernel,
+    target_os = "freebsd",
+    target_os = "netbsd",
+    target_os = "openbsd",
+    target_os = "dragonfly",
+    target_os = "illumos",
```

Comment:
Here and elsewhere, "freebsd" and "dragonfly" can use the config `freebsdlike`, while "netbsd" and "openbsd" can use the config `netbsdlike`, to shorten up the `cfg` code a little.